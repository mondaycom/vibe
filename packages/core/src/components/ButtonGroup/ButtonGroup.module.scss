@import "../../styles/typography";

.buttonGroup {
  display: flex;
  flex-direction: column;
}

.subTextContainer {
  @include font-caption();
}

.buttonsContainer {
  display: flex;
  align-items: center;
  justify-content: center;

  &.fullWidth {
    justify-content: flex-start;
    width: 100%;
  }

  .button {
    border-style: solid;
    border-width: 1px 0;
    background-color: var(--primary-background-color);
    border-color: var(--ui-border-color);

    // &:first-of-type {
    //   border-inline-start-width: 1px;
    // }

    // &:last-of-type {
    //   border-inline-end-width: 1px;
    // }

    &.activeButton {
      background-color: var(--primary-selected-color);
      border-color: var(--primary-color);
      border-inline-start-width: 1px;
      border-inline-end-width: 1px;

      &:hover {
        background-color: var(--primary-selected-color);
      }
    }

    &.disabled,
    &.buttonDisabled {
      background-color: var(--disabled-background-color);
      opacity: 0.5;
      cursor: not-allowed;
    }

    &.startBorder {
      border-inline-start-width: 1px;
    }

    &.endBorder {
      border-inline-end-width: 1px;
    }

    &.fullWidth {
      width: 100%;
      display: flex;
    }
  }

  .kindTertiary & {
    .button {
      border-color: transparent;

      &:first-child {
        border-color: transparent;
      }
    }
  }
}

.optionText:not(.disabled) {
  color: var(--primary-text-color);
}

.optionText:not(.disabled).selected {
  color: var(--primary-text-color);
}

.wrapper.disabled {
  cursor: not-allowed;
}

.fullWidth {
  flex: 1;
}
