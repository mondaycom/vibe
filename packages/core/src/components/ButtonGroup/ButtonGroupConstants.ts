export type ButtonValue = string | number;

/**
 * @deprecated
 */
export declare enum ButtonType {
  PRIMARY = "primary",
  SECONDARY = "secondary",
  TERTIARY = "tertiary"
}
