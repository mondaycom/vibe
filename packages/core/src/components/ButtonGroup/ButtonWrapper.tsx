import React from "react";
import { isNil } from "lodash-es";
import Button, { ButtonProps } from "../Button/Button";
import Tooltip from "../Tooltip/Tooltip";
import { MoveBy } from "../../types/MoveBy";
import { TooltipPositions } from "../Tooltip/Tooltip.types";

export interface ButtonWrapperProps extends ButtonProps {
  /**
   * The content of the tooltip.
   */
  tooltipContent?: string;
  /**
   * The position of the tooltip relative to the button.
   */
  tooltipPosition?: TooltipPositions;
  /**
   * The delay in milliseconds before the tooltip hides.
   */
  tooltipHideDelay?: number;
  /**
   * The delay in milliseconds before the tooltip shows.
   */
  tooltipShowDelay?: number;
  /**
   * CSS selector for the tooltip container.
   */
  tooltipContainerSelector?: string;
  /**
   * Adjusts the tooltip position.
   */
  tooltipMoveBy?: MoveBy;
}

export const ButtonWrapper = ({
  tooltipContent,
  tooltipPosition,
  tooltipHideDelay,
  tooltipShowDelay,
  tooltipContainerSelector,
  tooltipMoveBy,
  ...otherProps
}: ButtonWrapperProps) => {
  let button = <Button {...otherProps} />;
  if (!isNil(tooltipContent)) {
    button = (
      <Tooltip
        moveBy={tooltipMoveBy}
        position={tooltipPosition}
        hideDelay={tooltipHideDelay}
        showDelay={tooltipShowDelay}
        content={tooltipContent}
        showTrigger={["mouseenter"]}
        hideTrigger={["mouseleave"]}
        containerSelector={tooltipContainerSelector}
      >
        {button}
      </Tooltip>
    );
  }

  return button;
};
