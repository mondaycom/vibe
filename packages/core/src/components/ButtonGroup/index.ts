export { default as ButtonGroup, type ButtonGroupProps } from "./ButtonGroup";
