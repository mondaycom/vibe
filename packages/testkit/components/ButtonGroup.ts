import { test, Page, Locator } from "@playwright/test";
import { BaseElement } from "./BaseElement";
import { Button } from "./Button";

/**
 * Class representing a group of buttons.
 * Extends the BaseElement class.
 */
export class ButtonGroup extends BaseElement {
  constructor(page: Page, locator: Locator, elementReportName: string) {
    super(page, locator, elementReportName);
  }

  /**
   * Get all buttons in the button group.
   * @returns {Promise<Button[]>} An array of Button objects.
   */
  async getAllButtons(): Promise<Button[]> {
    let buttons: Button[] = [];
    await test.step(`Get all buttons in ${this.elementReportName}`, async () => {
      const buttonsLocators = await this.locator.locator("button").all();
      const buttonPromises = buttonsLocators.map(
        async buttonLocator => new Button(this.page, buttonLocator, await buttonLocator.innerText())
      );
      buttons = await Promise.all(buttonPromises);
    });
    return buttons;
  }

  /**
   * Get a button by its name.
   * @param {string} buttonName - The name of the button to retrieve.
   * @returns {Button} The button with the specified name.
   */
  async getButtonByName(buttonName: string): Promise<Button | undefined> {
    let button: Button | undefined;
    await test.step(`Get button by name ${buttonName} in ${this.elementReportName}`, async () => {
      button = new Button(
        this.page,
        this.locator.locator("button").filter({ hasText: buttonName }),
        `Button: ${buttonName}`
      );
    });
    return button;
  }

  /**
   * Click a button by its name.
   * @param {string} buttonName - The name of the button to click.
   * @returns {Promise<void>}
   */
  async clickButton(buttonName: string): Promise<void> {
    await test.step(`Click button ${buttonName} in ${this.elementReportName}`, async () => {
      const button = new Button(
        this.page,
        this.locator.locator("button").filter({ hasText: buttonName }),
        `Button: ${buttonName}`
      );
      if (!button) {
        throw new Error(`Invalid button name provided: ${buttonName}`);
      }
      await button.click();
    });
  }
}
