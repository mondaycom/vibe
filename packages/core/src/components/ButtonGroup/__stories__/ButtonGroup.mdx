import { Meta } from "@storybook/blocks";
import ButtonGroup from "../ButtonGroup";
import { Mobile } from "@vibe/icons";
import { BREADCRUBMS, BUTTON, TABS } from "../../../storybook/components/related-components/component-description-map";
import Desktop from "./assets/Desktop";
import { TipCheckYourself } from "./ButtonGroup.stories.helpers";
import * as ButtonGroupStories from "./ButtonGroup.stories";

<Meta of={ButtonGroupStories} />

# ButtonGroup

- [Overview](#overview)
- [Props](#props)
- [Usage](#usage)
- [Variants](#variants)
- [Do’s and don’ts](#dos-and-donts)
- [Use cases and examples](#use-cases-and-examples)
- [Related components](#related-components)
- [Feedback](#feedback)

## Overview

ButtonGroup can be used to group related options. To emphasize groups of related toggle buttons, a group should share a common container.

<Canvas of={ButtonGroupStories.Overview} />

### Import

```js
import { ButtonGroup } from "@vibe/core";
```

## Props

<PropsTable />

## Usage

<UsageGuidelines
  guidelines={[
    "Group together calls to action that are related to each other.",
    "Use when all the actions in the group take place immediately in one click. If one or more actions does not immediately happen in one click, either use a different component, or remove those actions from the group.",
    "Only should be used in groups of up to six buttons if the buttons contain an icon with no text.",
    "The main action of a button group will be in selected mode.",
    <>
      Keep the content short and actionable. For longer actions, use the{" "}
      <StorybookLink page="Components/Dropdown">Dropdown</StorybookLink> component.
    </>,
    <>
      Button group is used to display the same content in different view, if you need to present different content, use{" "}
      <StorybookLink page="Components/Tabs">Tabs.</StorybookLink>
    </>
  ]}
/>

<TipCheckYourself />

## Variants

### Default

<Canvas of={ButtonGroupStories.Default} />

### Tertiary

<Canvas of={ButtonGroupStories.Tertiary} />

### Disabled - all buttons

<Canvas of={ButtonGroupStories.Disabled} />

### Disabled - single button

<Canvas of={ButtonGroupStories.DisabledSingeButton} />

### Size

<Canvas of={ButtonGroupStories.Size} />

## Do’s and Don’ts

<ComponentRules
  rules={[
    {
      positive: {
        component: (
          <ButtonGroup
            groupAriaLabel="button group aria label"
            value={1}
            options={[
              { value: 1, text: "Month" },
              { value: 2, text: "Week" },
              { value: 3, text: "Year" }
            ]}
          />
        ),
        description: "Use when all the actions in the group take place immediately in one click."
      },
      negative: {
        component: (
          <ButtonGroup
            groupAriaLabel="button group aria label"
            value={1}
            options={[
              { value: 1, text: "Main table" },
              { value: 2, text: "Hadas view" },
              { value: 3, text: "Chart view" }
            ]}
          />
        ),
        description: (
          <>
            Don’t use a button group when switching between content or object pages. Use{" "}
            <StorybookLink page="Components/Tabs">Tabs</StorybookLink> instead.
          </>
        )
      }
    },
    {
      positive: {
        component: (
          <div className="monday-storybook-button-group_column">
            <ButtonGroup
              groupAriaLabel="button group aria label"
              size={ButtonGroup.sizes.MEDIUM}
              value={1}
              options={[
                { value: 1, leftIcon: Desktop },
                { value: 2, leftIcon: Mobile }
              ]}
            />
            <ButtonGroup
              groupAriaLabel="button group aria label"
              size={ButtonGroup.sizes.MEDIUM}
              value={1}
              options={[
                { value: 1, text: "Desktop" },
                { value: 2, text: "Mobile" }
              ]}
            />
            <ButtonGroup
              groupAriaLabel="button group aria label"
              size={ButtonGroup.sizes.MEDIUM}
              value={1}
              options={[
                { value: 1, text: "Desktop", leftIcon: Desktop },
                { value: 2, text: "Mobile", leftIcon: Mobile }
              ]}
            />
          </div>
        ),
        description: "Always use buttons with a consistent style: Icons, text or icon with text."
      },
      negative: {
        component: (
          <ButtonGroup
            groupAriaLabel="button group aria label"
            size={ButtonGroup.sizes.MEDIUM}
            value={1}
            options={[
              { value: 1, text: "Desktop" },
              { value: 2, icon: Mobile }
            ]}
          />
        ),
        description: "Avoid combining text buttons with icon buttons within the same button group."
      }
    },
    {
      positive: {
        component: (
          <ButtonGroup
            groupAriaLabel="button group aria label"
            value={1}
            options={[
              { value: 1, text: "Yearly plan" },
              { value: 2, text: "Monthly plan" },
              { value: 3, text: "Annual Plan" }
            ]}
          />
        ),
        description: "Keep button copy width as simillar as possible."
      },
      negative: {
        component: (
          <ButtonGroup
            groupAriaLabel="button group aria label"
            value={1}
            options={[
              { value: 1, text: "Yearly" },
              { value: 2, text: "Monthly" },
              { value: 3, text: "Annual-half year plan" }
            ]}
          />
        ),
        description: "Avoid mixing long button copy and short copy."
      }
    }
  ]}
/>

## Use cases and examples

### Button group in settings

For example: on the views settings you can choose only one option.

<Canvas of={ButtonGroupStories.ButtonGroupInSettings} />

### Button group as toggle

Use button group as toggle for change the view between two states.
For on and off actions, use the <StorybookLink page="Components/Toggle">Toggle</StorybookLink> component.

<Canvas of={ButtonGroupStories.ButtonGroupAsToggle} />

### Full width button group

Use this option in order for the ButtonGroup to fill the entire width of its container

This feature is intended to be used within small containers like a side panel, menu or similar small floating elements

<Canvas of={ButtonGroupStories.FullWidthButtonGroup} />

## Related components

<RelatedComponents componentsNames={[TABS, BUTTON, BREADCRUBMS]} />
