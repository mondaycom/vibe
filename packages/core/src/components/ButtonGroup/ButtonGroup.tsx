import React, { forwardRef, useCallback, useEffect, useMemo, useRef, useState } from "react";
import { camelCase } from "lodash-es";
import cx from "classnames";
import { getStyle } from "../../helpers/typesciptCssModulesHelper";
import Button from "../Button/Button";
import usePrevious from "../../hooks/usePrevious";
import useMergeRef from "../../hooks/useMergeRef";
import { ButtonValue } from "./ButtonGroupConstants";
import { ButtonWrapper } from "./ButtonWrapper";
import { SIZES } from "../../constants";
import { ButtonType as ButtonTypeEnum } from "../Button/ButtonConstants";
import { ButtonType, ButtonSize } from "../Button";
import { SubIcon, VibeComponentProps, withStaticProps } from "../../types";
import { MoveBy } from "../../types/MoveBy";
import { getTestId } from "../../tests/test-ids-utils";
import { ComponentDefaultTestId, ComponentVibeId } from "../../tests/constants";
import styles from "./ButtonGroup.module.scss";
import { TooltipPositions } from "../Tooltip";

type ButtonGroupOption = {
  icon?: SubIcon;
  leftIcon?: SubIcon;
  ariaLabel?: string;
  subText?: string;
  value: ButtonValue;
  text: string;
  disabled?: boolean;
  tooltipContent?: string;
};

export interface ButtonGroupProps extends VibeComponentProps {
  /**
   * The list of button options.
   */
  options: Array<ButtonGroupOption>;
  /**
   * The currently selected button value.
   */
  value?: ButtonValue;
  /**
   * Callback fired when a button is selected.
   */
  onSelect?: (value: ButtonValue, name: string) => void;
  /**
   * The size of the buttons.
   */
  size?: ButtonSize;
  /**
   * The style variant of the buttons.
   */
  kind?: Extract<ButtonType, "secondary" | "tertiary">;
  /**
   * The name of the button group.
   */
  name?: string;
  /**
   * If true, disables all buttons in the group.
   */
  disabled?: boolean;
  /**
   * The label of the button group for accessibility.
   */
  groupAriaLabel?: string;
  /**
   * The position of the tooltip relative to the button.
   */
  tooltipPosition?: TooltipPositions;
  /**
   * The delay in milliseconds before the tooltip hides.
   */
  tooltipHideDelay?: number;
  /**
   * The delay in milliseconds before the tooltip shows.
   */
  tooltipShowDelay?: number;
  /**
   * CSS selector for the tooltip container.
   */
  tooltipContainerSelector?: string;
  /**
   * Adjusts the tooltip position.
   */
  tooltipMoveBy?: MoveBy;
  /**
   * The content inside the button group.
   */
  children?: React.ReactNode;
  /**
   * If true, makes the button group take the full width of its container.
   */
  fullWidth?: boolean;
  /**
   * If true, removes focus from the button after clicking.
   */
  blurOnMouseUp?: boolean;
}

const ButtonGroup = forwardRef(
  (
    {
      className,
      options,
      name = "",
      disabled = false,
      value = "",
      onSelect,
      size = "small",
      kind = "secondary",
      groupAriaLabel = "",
      tooltipPosition,
      tooltipHideDelay,
      tooltipShowDelay,
      tooltipContainerSelector,
      tooltipMoveBy,
      blurOnMouseUp = true,
      id,
      "data-testid": dataTestId,
      fullWidth = false
    }: ButtonGroupProps,
    ref: React.ForwardedRef<HTMLDivElement>
  ) => {
    const inputRef = useRef();
    const mergedRef = useMergeRef(ref, inputRef);

    const [valueState, setValueState] = useState(value);
    const prevValue = usePrevious(value);

    const onClick = useCallback(
      (option: ButtonGroupOption) => {
        const isDisabled = disabled || option.disabled;
        if (!isDisabled) {
          setValueState(option.value);
          if (onSelect) {
            onSelect(option.value, name);
          }
        }
      },
      [onSelect, disabled, name]
    );

    const selectedOption = useMemo(() => {
      return options.find(option => option.value === valueState);
    }, [options, valueState]);

    const Buttons = useMemo(() => {
      return options.map((option, index) => {
        const isSelected = option.value === valueState;
        return (
          <ButtonWrapper
            key={option.value}
            size={size}
            onClick={() => onClick(option)}
            rightIcon={option.icon}
            leftIcon={option.leftIcon}
            active={isSelected}
            rightFlat={index !== options.length - 1}
            leftFlat={index !== 0}
            kind="tertiary"
            preventClickAnimation
            ariaLabel={option.ariaLabel}
            tooltipContent={option.tooltipContent}
            tooltipPosition={tooltipPosition}
            tooltipHideDelay={tooltipHideDelay}
            tooltipShowDelay={tooltipShowDelay}
            tooltipContainerSelector={tooltipContainerSelector}
            tooltipMoveBy={tooltipMoveBy}
            blurOnMouseUp={blurOnMouseUp}
            className={cx(styles.button, styles.optionText, {
              [styles.selected]: isSelected,
              [styles.disabled]: disabled,
              [styles.buttonDisabled]: option.disabled,
              [styles.fullWidth]: fullWidth
            })}
            activeButtonClassName={styles.activeButton}
          >
            {option.text}
          </ButtonWrapper>
        );
      });
    }, [
      options,
      valueState,
      size,
      tooltipPosition,
      tooltipHideDelay,
      tooltipShowDelay,
      tooltipContainerSelector,
      tooltipMoveBy,
      blurOnMouseUp,
      disabled,
      fullWidth,
      onClick
    ]);

    // Effects
    useEffect(() => {
      // Update value if changed from props
      if (value !== prevValue && value !== valueState) {
        setValueState(value);
      }
    }, [value, prevValue, valueState, setValueState]);

    return (
      <div
        className={cx(styles.buttonGroup, className, getStyle(styles, camelCase("kind-" + kind)), {
          [styles.disabled]: disabled
        })}
        id={id}
        data-testid={dataTestId || getTestId(ComponentDefaultTestId.BUTTON_GROUP, id)}
        data-vibe={ComponentVibeId.BUTTON_GROUP}
        ref={mergedRef}
      >
        <div role="group" aria-label={groupAriaLabel} className={cx(styles.buttonsContainer)} aria-disabled={disabled}>
          {Buttons}
        </div>
        {selectedOption && selectedOption.subText && (
          <div className={cx(styles.subTextContainer)}>{selectedOption.subText}</div>
        )}
      </div>
    );
  }
);

interface ButtonGroupStaticProps {
  sizes: typeof SIZES;
  kinds: typeof ButtonTypeEnum;
}

export default withStaticProps<ButtonGroupProps, ButtonGroupStaticProps>(ButtonGroup, {
  sizes: Button.sizes,
  kinds: Button.kinds
});
