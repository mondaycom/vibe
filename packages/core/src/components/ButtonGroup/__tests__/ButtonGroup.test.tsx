import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import ButtonGroup from "../ButtonGroup";

describe("ButtonGroup tests", () => {
  const option = [
    { value: 1, text: "Alpha" },
    { value: 2, text: "Beta" },
    { value: 3, text: "Gamma" },
    { value: 4, text: "Delta" }
  ];
  let onSelectMock: jest.Mock;
  let label: string;

  beforeEach(() => {
    onSelectMock = jest.fn();
    label = "label";
    render(<ButtonGroup options={option} onSelect={onSelectMock} groupAriaLabel={label} />);
  });

  it("should call the click callback when clicked", () => {
    fireEvent.click(screen.getByText("Delta"));
    expect(onSelectMock.mock.calls.length).toBe(1);
  });

  describe("a11y", () => {
    it("should add the label", () => {
      expect(screen.getByLabelText(label)).toBeTruthy();
    });
  });

  describe("snapshots", () => {
    const snapshotOptions = [
      { value: 1, text: "Alpha" },
      { value: 2, text: "Beta" },
      { value: 3, text: "Gamma" },
      { value: 4, text: "Delta" }
    ];

    it("should render ButtonGroup with full width", () => {
      const { container } = render(
        <ButtonGroup options={snapshotOptions} groupAriaLabel="Full width test" fullWidth={true} value={2} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it("should render ButtonGroup without tooltips", () => {
      const { container } = render(
        <ButtonGroup options={snapshotOptions} groupAriaLabel="No tooltips test" value={1} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it("should render ButtonGroup with tooltips on all options", () => {
      const optionsWithAllTooltips = [
        { value: 1, text: "Alpha", tooltipContent: "First option tooltip" },
        { value: 2, text: "Beta", tooltipContent: "Second option tooltip" },
        { value: 3, text: "Gamma", tooltipContent: "Third option tooltip" },
        { value: 4, text: "Delta", tooltipContent: "Fourth option tooltip" }
      ];

      const { container } = render(
        <ButtonGroup options={optionsWithAllTooltips} groupAriaLabel="All tooltips test" value={3} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it("should render ButtonGroup with tooltips on some options", () => {
      const optionsWithSomeTooltips = [
        { value: 1, text: "Alpha" },
        { value: 2, text: "Beta", tooltipContent: "Beta tooltip" },
        { value: 3, text: "Gamma" },
        { value: 4, text: "Delta", tooltipContent: "Delta tooltip" }
      ];

      const { container } = render(
        <ButtonGroup options={optionsWithSomeTooltips} groupAriaLabel="Some tooltips test" value={4} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
