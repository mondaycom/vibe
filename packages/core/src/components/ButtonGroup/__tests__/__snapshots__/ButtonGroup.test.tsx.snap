// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ButtonGroup tests > snapshots > should render ButtonGroup with full width 1`] = `
<div
  class="buttonGroup kindSecondary"
  data-testid="button-group"
  data-vibe="ButtonGroup"
>
  <div
    aria-disabled="false"
    aria-label="Full width test"
    class="buttonsContainer fullWidth"
    role="group"
  >
    <div
      class="fullWidth"
    >
      <button
        aria-busy="false"
        aria-disabled="false"
        class="button optionText fullWidth startBorder button sizeSmall kindTertiary colorPrimary rightFlat preventClickAnimation"
        data-testid="button"
        data-vibe="Button"
        type="button"
      >
        Alpha
      </button>
    </div>
    <div
      class="fullWidth"
    >
      <button
        aria-busy="false"
        aria-disabled="false"
        class="button optionText selected fullWidth endBorder button sizeSmall kindTertiary colorPrimary colorPrimaryActive activeButton rightFlat leftFlat preventClickAnimation"
        data-testid="button"
        data-vibe="Button"
        type="button"
      >
        Beta
      </button>
    </div>
    <div
      class="fullWidth"
    >
      <button
        aria-busy="false"
        aria-disabled="false"
        class="button optionText fullWidth endBorder button sizeSmall kindTertiary colorPrimary rightFlat leftFlat preventClickAnimation"
        data-testid="button"
        data-vibe="Button"
        type="button"
      >
        Gamma
      </button>
    </div>
    <div
      class="fullWidth"
    >
      <button
        aria-busy="false"
        aria-disabled="false"
        class="button optionText fullWidth endBorder button sizeSmall kindTertiary colorPrimary leftFlat preventClickAnimation"
        data-testid="button"
        data-vibe="Button"
        type="button"
      >
        Delta
      </button>
    </div>
  </div>
</div>
`;

exports[`ButtonGroup tests > snapshots > should render ButtonGroup with tooltips on all options 1`] = `
<div
  class="buttonGroup kindSecondary"
  data-testid="button-group"
  data-vibe="ButtonGroup"
>
  <div
    aria-disabled="false"
    aria-label="All tooltips test"
    class="buttonsContainer"
    role="group"
  >
    <span
      class=""
    >
      <button
        aria-busy="false"
        aria-disabled="false"
        class="button optionText startBorder endBorder button sizeSmall kindTertiary colorPrimary rightFlat preventClickAnimation"
        data-testid="button"
        data-vibe="Button"
        type="button"
      >
        Alpha
      </button>
    </span>
    <span
      class=""
    >
      <button
        aria-busy="false"
        aria-disabled="false"
        class="button optionText button sizeSmall kindTertiary colorPrimary rightFlat leftFlat preventClickAnimation"
        data-testid="button"
        data-vibe="Button"
        type="button"
      >
        Beta
      </button>
    </span>
    <span
      class=""
    >
      <button
        aria-busy="false"
        aria-disabled="false"
        class="button optionText selected endBorder button sizeSmall kindTertiary colorPrimary colorPrimaryActive activeButton rightFlat leftFlat preventClickAnimation"
        data-testid="button"
        data-vibe="Button"
        type="button"
      >
        Gamma
      </button>
    </span>
    <span
      class=""
    >
      <button
        aria-busy="false"
        aria-disabled="false"
        class="button optionText endBorder button sizeSmall kindTertiary colorPrimary leftFlat preventClickAnimation"
        data-testid="button"
        data-vibe="Button"
        type="button"
      >
        Delta
      </button>
    </span>
  </div>
</div>
`;

exports[`ButtonGroup tests > snapshots > should render ButtonGroup with tooltips on some options 1`] = `
<div
  class="buttonGroup kindSecondary"
  data-testid="button-group"
  data-vibe="ButtonGroup"
>
  <div
    aria-disabled="false"
    aria-label="Some tooltips test"
    class="buttonsContainer"
    role="group"
  >
    <button
      aria-busy="false"
      aria-disabled="false"
      class="button optionText startBorder endBorder button sizeSmall kindTertiary colorPrimary rightFlat preventClickAnimation"
      data-testid="button"
      data-vibe="Button"
      type="button"
    >
      Alpha
    </button>
    <span
      class=""
    >
      <button
        aria-busy="false"
        aria-disabled="false"
        class="button optionText endBorder button sizeSmall kindTertiary colorPrimary rightFlat leftFlat preventClickAnimation"
        data-testid="button"
        data-vibe="Button"
        type="button"
      >
        Beta
      </button>
    </span>
    <button
      aria-busy="false"
      aria-disabled="false"
      class="button optionText button sizeSmall kindTertiary colorPrimary rightFlat leftFlat preventClickAnimation"
      data-testid="button"
      data-vibe="Button"
      type="button"
    >
      Gamma
    </button>
    <span
      class=""
    >
      <button
        aria-busy="false"
        aria-disabled="false"
        class="button optionText selected endBorder button sizeSmall kindTertiary colorPrimary colorPrimaryActive activeButton leftFlat preventClickAnimation"
        data-testid="button"
        data-vibe="Button"
        type="button"
      >
        Delta
      </button>
    </span>
  </div>
</div>
`;

exports[`ButtonGroup tests > snapshots > should render ButtonGroup without tooltips 1`] = `
<div
  class="buttonGroup kindSecondary"
  data-testid="button-group"
  data-vibe="ButtonGroup"
>
  <div
    aria-disabled="false"
    aria-label="No tooltips test"
    class="buttonsContainer"
    role="group"
  >
    <button
      aria-busy="false"
      aria-disabled="false"
      class="button optionText selected startBorder endBorder button sizeSmall kindTertiary colorPrimary colorPrimaryActive activeButton rightFlat preventClickAnimation"
      data-testid="button"
      data-vibe="Button"
      type="button"
    >
      Alpha
    </button>
    <button
      aria-busy="false"
      aria-disabled="false"
      class="button optionText endBorder button sizeSmall kindTertiary colorPrimary rightFlat leftFlat preventClickAnimation"
      data-testid="button"
      data-vibe="Button"
      type="button"
    >
      Beta
    </button>
    <button
      aria-busy="false"
      aria-disabled="false"
      class="button optionText endBorder button sizeSmall kindTertiary colorPrimary rightFlat leftFlat preventClickAnimation"
      data-testid="button"
      data-vibe="Button"
      type="button"
    >
      Gamma
    </button>
    <button
      aria-busy="false"
      aria-disabled="false"
      class="button optionText endBorder button sizeSmall kindTertiary colorPrimary leftFlat preventClickAnimation"
      data-testid="button"
      data-vibe="Button"
      type="button"
    >
      Delta
    </button>
  </div>
</div>
`;
